app/scripts/directives/mobileIntegration.js

```diff
@@ -0,0 +1,122 @@
+              var bindingProviderName = _.get(binding, ['metadata', 'annotations', 'integrations.aerogear.org/provider']);
+              var bindingConsumerName = _.get(binding, ['metadata', 'annotations', 'integrations.aerogear.org/consumer']);
+              var consumerInstanceName = _.get(ctrl, 'consumerInstance.metadata.name');
+              var integrationServiceInstanceName = _.get(ctrl, 'integrationServiceInstance.metadata.name');
+              return (bindingProviderName && bindingConsumerName && consumerInstanceName && bindingProviderName === integrationServiceInstanceName && bindingConsumerName === consumerInstanceName);
+            });
+            ctrl.checkBinding();
+          }));
+        });
+
+        watches.push(DataService.watch(instancePreferredVersion, context, function(serviceInstancesData) {
+          var data = serviceInstancesData.by('metadata.name');
+          ctrl.checkIntegrationInstance(data);
+        }));
+      }
+    };
+
+    ctrl.checkIntegrationInstance = function(serviceInstances) {
+      ctrl.integrationServiceInstance = _.find(serviceInstances, function(serviceInstance) {
+        var clusterServiceClassExternalName = _.get(serviceInstance, 'spec.clusterServiceClassExternalName');
+        return (clusterServiceClassExternalName === ctrl.integrationServiceClass.spec.externalName);
+      });
+
+      ctrl.integrationInstanceProvisioning = _.get(ctrl, 'integrationServiceInstance.status.currentOperation') === 'Provision';
+      ctrl.integrationInstanceDeprovisioning = _.get(ctrl, 'integrationServiceInstance.status.currentOperation') === 'Deprovision';
+
+      ctrl.bindingMeta = {
+        annotations: {
+          'integrations.aerogear.org/consumer': _.get(ctrl, 'consumerInstance.metadata.name'),
+          'integrations.aerogear.org/provider': _.get(ctrl, 'integrationServiceInstance.metadata.name')
+        }
+      };
+    };
+
+    ctrl.checkBinding = function() {
+      if (ctrl.binding && ctrl.binding.status.currentOperation === 'Bind') {
+        ctrl.hasBinding = false;
+        ctrl.isBindPending = true;
+      }
+      if (ctrl.binding && isBindingReady(ctrl.binding)) {
+        ctrl.hasBinding = true;
+        ctrl.isBindPending = false;
+      }
+      if (!ctrl.binding) {
+        ctrl.hasBinding = false;
+        ctrl.isBindPending = false;
+      }
+      if (ctrl.binding && ctrl.binding.status.currentOperation === 'Unbind') {
+        ctrl.hasBinding = false;
+        ctrl.isBindPending = true;
+      }
+    };
+
+    ctrl.integrationPanelVisible = false;
+
+    ctrl.closeIntegrationPanel = function() {
+      ctrl.integrationPanelVisible = false;
+    };
+
+    ctrl.openIntegrationPanel = function() {
+      ctrl.integrationPanelVisible = true;
+    };
+
+    ctrl.provision = function() {
+      $scope.$emit('open-overlay-panel', Catalog.getServiceItem(ctrl.integrationServiceClass));
+    };
+
+    ctrl.$onDestroy = function() {
```

Comment:
ahh i see we are removing them here, but still wondering about the pattern above where we add a watch on each change?